pkg/executer/executer.go

```diff
@@ -22,17 +37,90 @@ func NewExecuter() Executer {
 	return &executer{}
 }
 
-func (e *executer) Execute(command string, args ...string) (string, error) {
-	var stdoutBytes, stderrBytes bytes.Buffer
-	cmd := exec.Command(command, args...)
-	cmd.Stdout = &stdoutBytes
-	cmd.Stderr = &stderrBytes
-	err := cmd.Run()
+func (e *executer) Execute(command Command) (result string, err error) {
+	if len(command.Args) < 1 {
+		err = errors.New("at least one command line argument is required")
+		return
+	}
+	bin := command.Args[0]
+	args := command.Args[1:]
+	cmd := exec.Command(bin, args...)
+	if command.Env != nil {
+		cmd.Env = e.mergeEnv(os.Environ(), command.Env)
+	}
+	stdout := &bytes.Buffer{}
+	stderr := &bytes.Buffer{}
+	cmd.Stdout = stdout
+	cmd.Stderr = stderr
+	err = cmd.Run()
 	if err != nil {
-		return "", errors.Wrapf(err, "Failed to execute cmd (%s): %s", cmd, stderrBytes.String())
+		err = errors.Wrapf(err, "failed to execute cmd (%s): %s", bin, stderr.String())
+		return
+	}
+	result = strings.TrimSuffix(stdout.String(), "\n")
+	return
+}
+
+// mergeEnv merges the given environment with the given variables. The environment is expected to be
+// a list of strings in the the same format used by the os.Environ function. The variables is a map
+// describing the changes to make to that environment. The keys of the map are the names of the
+// variables and the values of the map the values. Variables with value nil will be removed from the
+// environment. The rest will be added, replacing any previous value.
+func (e *executer) mergeEnv(env []string, vars map[string]any) []string {
+	values := e.parseEnv(env)
+	for name, value := range vars {
+		if value == nil {
+			delete(values, name)
+		} else {
+			values[name] = fmt.Sprintf("%s", value)
+		}
+	}
+	return e.renderEnv(values)
+}
+
+// parseEnv takes a set of environment variables in the format used by os.Environ and puts them in a
+// map where the keys of the map are the names of the variables and the values of the map the values
+// of the variables.
+func (e *executer) parseEnv(env []string) map[string]string {
```

Comment:
Maybe we can use [godotenv](https://github.com/joho/godotenv) for it? I.e. godotenv.Unmarshal